rust/src/storage/file.rs

```diff
@@ -58,26 +65,110 @@ impl StorageBackend for FileStorageBackend {
     }
 
     async fn put_obj(&self, path: &str, obj_bytes: &[u8]) -> Result<(), StorageError> {
-        let dir = std::path::Path::new(path);
-
-        if let Some(d) = dir.parent() {
-            fs::create_dir_all(d).await?;
-        }
+        let tmp_file = self
+            .root
+            .join("_tmp_writes")
+            .join(Uuid::new_v4().to_string());
```

Comment:
I chose the relative path within table instead of env tmp (`/tmp`) because `rename(2)` will fail if source and destination will be at different partitions

Reply:
yeap, that's more I like it